enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,580 @@
+---
+title: assisted-installer-in-cluster
+authors:
+  - "@mhrivnak"
+  - "@hardys"
+  - "@avishayt"
+reviewers:
+  - "@eparis"
+  - "@markmc"
+  - "@dgoodwin"
+  - "@ronniel1"
+
+approvers:
+  - TBD
+creation-date: 2020-12-22
+last-updated: 2021-02-10
+status: implementable
+see-also:
+  - enhancements/installer/connected-assisted-installer.md
+  - enhancements/installer/assisted-installer-bare-metal-validations.md
+---
+
+# Assisted Installer in-cluster
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Assisted Installer currently runs as a SaaS on cloud.redhat.com, enabling
+users to deploy OpenShift clusters with certain customizations, particularly on
+bare metal hardware. It is necessary to bring those capabilities on-premise in
+users' "Hub" clusters by installing clusters via Multi-cluster management, such
+as through Hive and RHACM.
+
+This enhancement proposes the first iteration of running the Assisted Installer
+in end-user clusters to meet the purpose above.
+
+## Motivation
+
+The Assisted Installer running as a SaaS has demonstrated a new and valuable
+way to install OpenShift clusters through an agent-based flow. The [Assisted
+Installer for Connected Environments](connected-assisted-installer.md)
+enhancement proposal elaborates on the value of the agent-based installations.
+
+However many customers do not want a third party service running outside of
+their network to be able to provision hosts inside their network. Reasons
+include trust, control, and reproducibility. Those users will prefer to run the
+service on-premises. There are also many customers whose network environments
+have limited or no connectivity to the internet. Finally, it is desirable to
+integrate this new installation method into RHACM, which runs on-premesis.
```

Comment:
This last sentence takes as given that we want to replace the existing IPI implementation in RHACM with this new thing. I'm sure we've justified that decision somewhere, though? That justification should be summarized here, too, especially since the feature being replaced is GA already.

Reply:
This proposal only asserts that RHACM, an important stakeholder, wants to include an agent-based install flow. Whether they as a separate product choose to keep two bare metal install flows or replace bare metal IPI with the agent-based flow feels orthogonal to this proposal. The point here is just that RHACM is Red Hat's on-premises multi-cluster management product, agent-based installation is valuable on-premises, and RHACM wants to deliver this agent-based installation flow.